src/utils/cosmosdb_model_versioned.ts

```diff
@@ -166,6 +172,66 @@ export abstract class CosmosdbModelVersioned<
     );
   }
 
+  /**
+   * Add the given ttl at all the version of the document identified by
+   * the searchKey
+   */
+
+  public updateTTLForAllVersion(
+    searchKey: DocumentSearchKey<TR, ModelIdKey, PartitionKey>,
+    ttl: NonNegativeNumber
+  ): TE.TaskEither<CosmosErrors, ReadonlyArray<TR>> {
+    return pipe(
+      this.findAllVersionsByPartitionKey(searchKey),
+      TE.map(RA.rights),
+      TE.chain(documentList =>
+        RA.sequence(TE.ApplicativeSeq)(
+          documentList.map(document =>
+            super.updateTTL(
+              ([document.id, searchKey[0]] as unknown) as DocumentSearchKey<
+                TR,
+                CosmosDocumentIdKey,
+                PartitionKey
+              >,
+              ttl
+            )
+          )
+        )
+      ),
+      TE.mapLeft(toCosmosErrorResponse)
+    );
+  }
+
+  /**
+   * Given a searchKey returns all the version of a document
+   */
+
+  public findAllVersionsByPartitionKey(
+    searchKey: DocumentSearchKey<TR, ModelIdKey, PartitionKey>
+  ): TE.TaskEither<CosmosErrors, ReadonlyArray<t.Validation<TR>>> {
```

Comment:
Since we want to seach by partition key, maybe we can simplify this signature a little by moving from searchKey to partionKey parameter only.
Same suggestion for `updateTTLForAllVersion`

Reply:
As said, it's better to leave `seachKey` here. Sorry, my bad :)